docs/content/config/security/ssl.md

```diff
@@ -158,6 +158,81 @@ Certbot provisions certificates to `/etc/letsencrypt`. Add a volume to store the
 
     Certbot does support [alternative certificate providers via the `--server`][certbot::custom-ca] option. In most cases you'll want to use the default _Let's Encrypt_.
 
+#### Example using `certbot-dns-cloudflare` with Docker { data-toc-label='certbot-dns-cloudflare with Docker' }
+
+If for some reason, you can't expose your mail server's port 80 and port 443 (known as the `HTTP-01` challenge), you can consider obtaining a certificate by the `DNS-01` challenge.
+
+First, you need to obtain a Cloudflare API token
+
+1. Login into your Cloudflare dashboard.
+2. Navigate to MyProfile -> API Tokens.
+3. Create token a using the `Edit zone DNS` template since certbot [requires](https://certbot-dns-cloudflare.readthedocs.io/en/stable/#credentials) `ZONE:DNS:Edit` permission.
+
+!!! Zone resource configuration
+
+    Remember to choose a specific TLD, and create a token that allows access to all zone's resources is insecure here.
+
+4. Create `cloudflare.ini` with the following content:
+
+    ```dosini
+    dns_cloudflare_api_token = YOUR_CLOUDFLARE_TOKEN_HERE
+    ```
+
+5. Save it to the `.secrets` folder or anywhere you like. (Make sure to restrict its permission with the `chmod 600` command
+6. Add the following code into `docker-compose.yml`
+
+    ```yaml
+    services:
+      mailserver:
+        environments:
+          # Set SSL certificate type.
+          - SSL_TYPE=letsencrypt
+        volumes:
+          # We need to mount the cert generated by certbot into mail server.
+          - ./docker-data/certbot/certs/:/etc/letsencrypt
+      certbot-cloudflare:
+        image: certbot/dns-cloudflare:latest
+        container_name: certbot-cloudflare
+        volumes:
+          - ./docker-data/certbot/certs/:/etc/letsencrypt/
+          - ./docker-data/certbot/logs/:/var/log/letsencrypt/
+          - ./.secrets:/.secrets
+        command: certonly --dns-cloudflare --dns-cloudflare-credentials /.secrets/certbot/cloudflare.ini -d mail.example.com
+    ```
+
+    Alternative using the `docker run` command
+
+      ```sh
+      docker run \
+        --volume "${PWD}/docker-data/certbot/certs/:/etc/letsencrypt/" \
+        --volume "${PWD}/docker-data/certbot/logs/:/var/log/letsencrypt/" \
+        --volume "${PWD}/.secrets:/.secrets"
+        certbot/dns-cloudflare \
+        certonly --dns-cloudflare --dns-cloudflare-credentials /.secrets/certbot/cloudflare.ini -d mail.example.com
+      ```
```

Comment:
Apart from changing the list number to `5`, Revised a volume mount comment + `container_name` was removed.

Additionally changed the `docker-compose` config to use the [`secrets` feature](https://docs.docker.com/compose/compose-file/#secrets), although without Swarm only the file mount mode is available, and the long syntax features to adjust permissions and ownership don't work.. so I'm not sure how helpful that is as it's effectively a volume bind mount :sweat_smile: (_the `docker run` example also can't use it_).

So I'm open to dropping the `secrets` change, although I prefer it for being more explicit about content and separated from other volume mounts. The concerns about those gotcha's have been inlined into the docs example.

```suggestion
5. Your `docker-compose.yml` should include the following:

    ```yaml
    services:
      mailserver:
        environments:
          # Set SSL certificate type.
          - SSL_TYPE=letsencrypt
        volumes:
          # Mount the cert folder generated by Certbot into mail-server:
          - ./docker-data/certbot/certs/:/etc/letsencrypt/:ro

      certbot-cloudflare:
        image: certbot/dns-cloudflare:latest
        command: certonly --dns-cloudflare --dns-cloudflare-credentials /run/secrets/cloudflare-api-token -d mail.example.com
        volumes:
          - ./docker-data/certbot/certs/:/etc/letsencrypt/
          - ./docker-data/certbot/logs/:/var/log/letsencrypt/
        secrets:
          - cloudflare-api-token

    # Docs: https://docs.docker.com/engine/swarm/secrets/#use-secrets-in-compose
    # WARNING: In compose configs without swarm, the long syntax options have no effect,
    # Ensure that you properly `chmod 600` and `chown 0:0` the file on disk. Effectively treated as a bind mount.
    secrets:
      cloudflare-api-token:
        file: ./docker-data/certbot/secrets/cloudflare.ini
    ```

    Alternative using the `docker run` command (`secrets` feature is not available):

      ```sh
      docker run \
        --volume "${PWD}/docker-data/certbot/certs/:/etc/letsencrypt/" \
        --volume "${PWD}/docker-data/certbot/logs/:/var/log/letsencrypt/" \
        --volume "${PWD}/docker-data/certbot/secrets/:/tmp/secrets/certbot/"
        certbot/dns-cloudflare \
        certonly --dns-cloudflare --dns-cloudflare-credentials /tmp/secrets/certbot/cloudflare.ini -d mail.example.com
      ```
```